CMakeLists.txt

```diff
@@ -302,6 +302,14 @@ set(PADDLE_PYTHON_BUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}/python/build")
 set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -g -DNDEBUG")
 set(CMAKE_C_FLAGS_RELWITHDEBINFO "-O3 -g -DNDEBUG")
 
+if (ON_INFER)
+    message(STATUS "On inference mode, will take place some specific optimization.")
+    add_definitions(-DPADDLE_ON_INFERENCE)
+else()
+    #TODO(luotao), combine this warning with `make inference_lib_dist` command.
+    message(WARNING "On inference mode, will take place some specific optimization. Turn on the ON_INFER flag when building inference_lib only.")
+endif()
+
```

Comment:
This code is partially present in the main CMakeLists.txt file:
https://github.com/PaddlePaddle/Paddle/blob/f4c869d872a62d99cfbbd3e3c5c5d0cf2db4d863/CMakeLists.txt#L316-L321

Is it necessary to copy just to set an additional define?

Reply:
I do not quite understand.  I moved quoted section from very bottom of main CMakeLists.txt to a bit above in the same CMakeLists.txt. I cannot see any redundancy here

Reply:
@tpatejko These codes are only moved, not copied.

Reply:
My mistake. I haven't noticed that.